libobs/audio-monitoring/nix/pulseaudio-wrapper.h

```diff
@@ -0,0 +1,178 @@
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <inttypes.h>
+#include <pulse/stream.h>
+#include <pulse/context.h>
+#include <pulse/introspect.h>
+
+#pragma once
+
+struct pulseaudio_default_output {
+	char *default_sink_name;
+};
+
+struct enum_cb {
+	obs_enum_audio_device_cb cb;
+	void *data;
+	int cont;
+};
+
+void get_default_id(char **id);
+
+bool devices_match(const char *id1, const char *id2);
+
+/**
+ * Initialize the pulseaudio mainloop and increase the reference count
+ */
+int_fast32_t pulseaudio_init();
+
+/**
+ * Unreference the pulseaudio mainloop, when the reference count reaches
+ * zero the mainloop will automatically be destroyed
+ */
+void pulseaudio_unref();
+
+/**
+ * Lock the mainloop
+ *
+ * In order to allow for multiple threads to use the same mainloop pulseaudio
+ * provides it's own locking mechanism. This function should be called before
+ * using any pulseaudio function that is in any way related to the mainloop or
+ * context.
+ *
+ * @note use of this function may cause deadlocks
+ *
+ * @warning do not use with pulseaudio_ wrapper functions
+ */
+void pulseaudio_lock();
+
+/**
+ * Unlock the mainloop
+ *
+ * @see pulseaudio_lock()
+ */
+void pulseaudio_unlock();
+
+/**
+ * Wait for events to happen
+ *
+ * This function should be called when waiting for an event to happen.
+ */
+void pulseaudio_wait();
+
+/**
+ * Wait for accept signal from calling thread
+ *
+ * This function tells the pulseaudio mainloop wheter the data provided to
+ * the callback should be retained until the calling thread executes
+ * pulseaudio_accept()
+ *
+ * If wait_for_accept is 0 the function returns and the data is freed.
+ */
+void pulseaudio_signal(int wait_for_accept);
+
+/**
+ * Signal the waiting callback to return
+ *
+ * This function is used in conjunction with pulseaudio_signal()
+ */
+void pulseaudio_accept();
+
+/**
+ * Request source information
+ *
+ * The function will block until the operation was executed and the mainloop
+ * called the provided callback function.
+ *
+ * @return negative on error
+ *
+ * @note The function will block until the server context is ready.
+ *
+ * @warning call without active locks
+ */
+int_fast32_t
+pulseaudio_get_source_info_list(pa_source_info_cb_t cb, void *userdata);
+
+/**
+ * Request source information from a specific source
+ *
+ * The function will block until the operation was executed and the mainloop
+ * called the provided callback function.
+ *
+ * @param cb pointer to the callback function
+ * @param name the source name to get information for
+ * @param userdata pointer to userdata the callback will be called with
+ *
+ * @return negative on error
+ *
+ * @note The function will block until the server context is ready.
+ *
+ * @warning call without active locks
+ */
+int_fast32_t
```

Comment:
Style: Don't line break between the function return type and the function name in a function declaration.